src/go/k8s/api/redpanda/v1alpha2/common.go

```diff
@@ -0,0 +1,181 @@
+	// +optional
+	Password SecretKeyRef `json:"passwordSecretRef,omitempty"`
+	// Specifies the SASL/SCRAM authentication mechanism.
+	Mechanism SASLMechanism `json:"mechanism"`
+	// +optional
+	OAUth KafkaSASLOAuthBearer `json:"oauth,omitempty"`
+	// +optional
+	GSSAPIConfig KafkaSASLGSSAPI `json:"gssapi,omitempty"`
+	// +optional
+	AWSMskIam KafkaSASLAWSMskIam `json:"awsMskIam,omitempty"`
+}
+
+type SASLMechanism string
+
+const (
+	SASLMechanismPlain                  SASLMechanism = config.SASLMechanismPlain
+	SASLMechanismScramSHA256            SASLMechanism = config.SASLMechanismScramSHA256
+	SASLMechanismScramSHA512            SASLMechanism = config.SASLMechanismScramSHA512
+	SASLMechanismGSSAPI                 SASLMechanism = config.SASLMechanismGSSAPI
+	SASLMechanismOAuthBearer            SASLMechanism = config.SASLMechanismOAuthBearer
+	SASLMechanismAWSManagedStreamingIAM SASLMechanism = config.SASLMechanismAWSManagedStreamingIAM
+)
+
+// KafkaSASLOAuthBearer is the config struct for the SASL OAuthBearer mechanism
+type KafkaSASLOAuthBearer struct {
+	Token SecretKeyRef `json:"tokenSecretRef"`
+}
+
+// KafkaSASLGSSAPI represents the Kafka Kerberos config.
+type KafkaSASLGSSAPI struct {
+	AuthType           string       `json:"authType"`
+	KeyTabPath         string       `json:"keyTabPath"`
+	KerberosConfigPath string       `json:"kerberosConfigPath"`
+	ServiceName        string       `json:"serviceName"`
+	Username           string       `json:"username"`
+	Password           SecretKeyRef `json:"passwordSecretRef"`
+	Realm              string       `json:"realm"`
+
+	// EnableFAST enables FAST, which is a pre-authentication framework for Kerberos.
+	// It includes a mechanism for tunneling pre-authentication exchanges using armored KDC messages.
+	// FAST provides increased resistance to passive password guessing attacks.
+	EnableFast bool `json:"enableFast"`
+}
+
+// KafkaSASLAWSMskIam is the config for AWS IAM SASL mechanism,
+// see: https://docs.aws.amazon.com/msk/latest/developerguide/iam-access-control.html
+type KafkaSASLAWSMskIam struct {
+	AccessKey string       `json:"accessKey"`
+	SecretKey SecretKeyRef `json:"secretKeySecretRef"`
+
+	// SessionToken, if non-empty, is a session / security token to use for authentication.
+	// See: https://docs.aws.amazon.com/STS/latest/APIReference/welcome.html
+	SessionToken SecretKeyRef `json:"sessionTokenSecretRef"`
+
+	// UserAgent is the user agent to for the client to use when connecting
+	// to Kafka, overriding the default "franz-go/<runtime.Version()>/<hostname>".
+	//
+	// Setting a UserAgent allows authorizing based on the aws:UserAgent
+	// condition key; see the following link for more details:
+	// https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_condition-keys.html#condition-keys-useragent
+	UserAgent string `json:"userAgent"`
+}
+
+// CommonTLS specifies TLS configuration settings for Redpanda clusters that have authentication enabled.
+type CommonTLS struct {
```

Comment:
Renamed this `CommonTLS` from `KafkaTLS` since the parameters are generic enough to be used for initializing a client for any API. The `KafkaAPISpec` and `AdminAPISpec` essentially just differ in two main things:

1. Terminology around "brokers" v. "urls".
2. The SASL mechanisms they support for authenticating requests.